src/cargo/core/resolver/mod.rs

```diff
@@ -369,21 +403,32 @@ impl<T> RcVecIter<T> {
     }
 }
 
+impl<T> Clone for RcVecIter<T> {
+    fn clone(&self) -> RcVecIter<T> {
+        RcVecIter {
+            vec: self.vec.clone(),
+            rest: self.rest.clone(),
+        }
+    }
+}
+
 impl<T> Iterator for RcVecIter<T> where T: Clone {
     type Item = (usize, T);
+
```

Comment:
As far as I can tell, the only reason we return `(usize, T)` here is because `activate_deps_loop` uses it for debugging (trace-level). Is it worth keeping? It's arguably equivalent to just storing `Enumerate<RcVecIter<T>>`.